src/services/permissions/study.ts

```diff
@@ -34,21 +34,28 @@ export const canReadStudy = async (
     }
   }
 
-  let allowedStudies: Exclude<UserWithAllowedStudies, null>['allowedStudies']
+  let allowedStudiesId: string[]
   if ('allowedStudies' in user) {
-    allowedStudies = user.allowedStudies
+    allowedStudiesId = [
+      ...user.allowedStudies.map((allowedStudy) => allowedStudy.studyId),
+      ...user.contributors.map((contributor) => contributor.studyId),
+    ]
   } else {
```

Comment:
Je mettrais le 
```
 return allowedStudiesId.some((allowedStudiesId) => allowedStudiesId === study.id)
``` 
dans ce if, puis dans le else faire un `findAllowedStudy(studyId, userId) qui a chercher une étude qui directement ou via contributor est associée à l'utilisateur

Reply:
j'ai pas compris

Reply:
```suggestion
  if ('allowedStudies' in user) {
    const allowedStudiesId = [
      ...user.allowedStudies.map((allowedStudy) => allowedStudy.studyId),
      ...user.contributors.map((contributor) => contributor.studyId),
    ]

    if (allowedStudiesId.some((allowedStudiesId) => allowedStudiesId === study.id)) {
      return true
    }

    return false
  } else {
    const isAllowedUserOnStudy = await getAllowedUserOnStudy(user.id, study.id)
    return isAllowedUserOnStudy
  }
```

getAllowedUserOnStudy serait une fonction qui va cherche si l'utilisateur est associé à l'étude, directement ou via contributor.

Ça me parait plus efficace qu'aller chercher toutes les études de la personne et vérifier si celle qui est concernée est dedans

Reply:
Non, je recherche un objet, et la permission c'est l'existence ou non de cet objet. Exactement ce qui est fait là, mais on recherche un seul objet plutôt qu'une liste + recherche dans cette liste

Reply:
Et tu n'as pas le meme code selon ce que tu passes en entrée de ta fonction du coup ?